designsafe/static/scripts/data-depot/components/projects/project-listing/project-listing.component.js

```diff
@@ -18,9 +19,27 @@ function ProjectListingCtrl($scope, $state, DataBrowserService, Django, ProjectS
     ProjectService.list({offset:offset, limit:limit}).then(function(projects) {
       $scope.ui.busy = false;
       $scope.data.projects = _.map(projects, function(p) { p.href = $state.href('projects.view', {projectId: p.uuid}); return p; });
-      DataBrowserService.projectBreadcrumbSubject.next()
+      DataBrowserService.projectBreadcrumbSubject.next();
+      $scope.getNames();
     });
 
+    $scope.getNames = function () {
```

Comment:
I think this function should send over an array of usernames as a payload to the `getPublic` call. This may require some rewriting, but it is more extensible than using separators.

Reply:
That shouldn't be a big change though. I'll try passing the usernames array as something like `JSON.stringify(usernames)`? I can poke around on it tomorrow.